### PR TITLE
TBR formula fix

### PR DESCRIPTION
**Pull request by RiccardoM**
## Checklist
- [x] Targeted PR against correct branch
- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote tests
- [x] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For Admin Use:
- [x] Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- [x] Reviewers Assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]"

### CODE REVIEW

File: x/tbr/genesis.go [hunk 1]
```diff
@@ -2,48 +2,46 @@ package tbr
...
+	keeper.SetYearNumber(ctx, data.YearNumber)
 }
 
 // ExportGenesis returns a GenesisState for a given context and keeper.
 func ExportGenesis(ctx sdk.Context, keeper Keeper) GenesisState {
 	return GenesisState{
-		RewardDenom: keeper.GetRewardDenom(ctx),
-		PoolAmount:  keeper.GetBlockRewardsPool(ctx),
+		PoolAmount: keeper.GetTotalRewardPool(ctx),
+		YearNumber: keeper.GetYearNumber(ctx),
```
Comment (bragaz) [inline x/tbr/genesis.go]: Here you are not exporting `YearlyPoolAmount` and above in the `InitGenesis` its value will be `nil` everytime, so the `YearlyPoolAmount` will always be set with `data.PoolAmount`. I don't know if this was done by will or not. I'll let this comment here to let you know if maybe you didn't want to do this.

File: x/tbr/genesis.go [hunk 2]
```diff
@@ -2,48 +2,46 @@ package tbr
...
 
 // ValidateGenesis performs basic validation of genesis data returning an
 // error for any failed validation criteria.
 func ValidateGenesis(data GenesisState) error {
-	if len(strings.TrimSpace(data.RewardDenom)) == 0 {
-		return errors.New("transaction block reward reward denom cannot be empty")
-	}
-
 	if data.PoolAmount.Empty() {
 		return errors.New("transaction block reward pool cannot be empty")
```
Comment (bragaz) [inline x/tbr/genesis.go]: You shouldn't do some checks for the `YearNumber` data as well and possibly for the `YearlyPoolAmount` data ?

File: x/tbr/genesis.go (hunk 1 above)
Comment (RiccardoM) [reply x/tbr/genesis.go]: Good catch, I didn't see it. Added the export :rocket:

File: x/tbr/genesis.go (hunk 2 above)
Comment (RiccardoM) [reply x/tbr/genesis.go]: I'm fine with having the default year number to `0` initially. Good catch for the missed `YearlyPoolReward` value checks. Added!

File: x/tbr/internal/keeper/keeper.go [hunk 3]
```diff
@@ -13,153 +12,193 @@ import (
...
+// ---------------------------
 
-	// Compute his validation power
-	validatorPower := validatorStakedTokens.Quo(totalStakedTokens.ToDec())
+// ComputeProposerReward computes the final reward for the validator block's proposer
+func (k Keeper) ComputeProposerReward(ctx sdk.Context, validatorsCount int64,
+	proposer exported.ValidatorI, totalStakedTokens sdk.Int) sdk.DecCoins {
+
+	// Get the maximum year reward by multiplying the yearly pool by V/100
+	Ry := k.GetYearlyRewardPool(ctx).MulDec(sdk.NewDec(validatorsCount)).QuoDec(sdk.NewDec(100))
```
Comment (marcotradenet) [inline x/tbr/internal/keeper/keeper.go]: I think that If you use `GetYearlyRewardPool` the formula doesn't work properly: every `ComputeProposerReward` iteration gets amount under the yearly reward pool target.  In discussed formula the value for `Ry` is `Py x V/100` and it is only function of year changes and number of validators.

File: x/tbr/internal/keeper/keeper.go [hunk 4]
```diff
@@ -13,153 +12,193 @@ import (
...
-	vNumber := sdk.NewDecWithPrec(validatorsNumber, 0)
+	BPY = DPY.Mul(HPD).Mul(MPH).Mul(BPM) // Blocks Per Year
+)
 
-	secondMember := averageValidatorsNumber.Quo(vNumber)
+func computeYearFromBlockHeight(blockHeight int64) int64 {
+	// Divide the current block number to the number of blocks per year to get the year value
+	// Truncate the result so that 1.99 years = 1 year and not 2
+	blocksPerYear := DPY.Mul(HPD).Mul(MPH).Mul(BPM)
+	return sdk.NewDec(blockHeight).Quo(blocksPerYear).TruncateInt64()
```
[the remark this answers is not in the document]
Comment (RiccardoM) [reply x/tbr/internal/keeper/keeper.go]: Thanks for the tip, I've implemented it this way now.

File: x/tbr/internal/keeper/keeper.go (hunk 3 above)
Comment (RiccardoM) [reply x/tbr/internal/keeper/keeper.go]: You're right. I've revisited the code with the following changes: 

- Added the `YearlyPoolRemains` which tells how many coins are left for the current year
- Now the yearly pool is no more decremented when distributing the rewards. Only the `TotalRewardPool` and the `YearlyPoolRemains` are

**Pull request merged**